packages/core/src/routes/session/passwordless.ts

```diff
@@ -148,195 +154,103 @@ export default function passwordlessRoutes<T extends AnonymousRouter>(
     }
   );
 
-  router.post(
-    `${signInRoute}/sms/send-passcode`,
-    koaGuard({ body: object({ phone: string().regex(phoneRegEx) }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { phone } = ctx.guard.body;
-      const type = 'SignInSmsSendPasscode';
-      ctx.log(type, { phone });
+  router.post(`${signInRoute}/sms`, async (ctx, next) => {
+    const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+    const verificationStorage = parseVerificationStorage(result);
 
-      assertThat(
-        await hasUserWithPhone(phone),
-        new RequestError({ code: 'user.phone_not_exists', status: 422 })
-      );
+    const { phone, flow, expiresAt } = verificationStorage;
+    assertThat(phone, new RequestError({ code: 'session.passwordless_not_verified', status: 401 }));
```

Comment:
nit: should we be more specific on the error, like session.passwordless_not_verified, status: 401, method: phone. 
so phrases can contain the failed method type phone here.

Reply:
complementary info added to error constructors.